JS.md

```diff
@@ -0,0 +1,237 @@
+# JavaScript API
+
+In the [MVP](MVP.md), the only way to access WebAssembly on the Web is through
+an explicit JS API which is defined below. (In the future, WebAssembly may also
+be loaded and run directly from an HTML `<script type='module'>` tag—and
+any other Web API that loads ES6 modules via URL—as part of 
+[ES6 Module integration](Modules.md#integration-with-es6-modules).)
+
+## The `Wasm` object
+
+The `Wasm` object is the initial value of the `Wasm` property of the global
+object. Like the `Math` and `JSON` objects, the `Wasm` object is a plain
+JS object (not a constructor or function) that acts like a namespace and has the
+following properties:
+
+### Constructor Properties of the `Wasm` object
+
+* `Wasm.Module` : see [`Wasm.Module` objects](#wasmmodule-objects)
+* `Wasm.Instance` : see [`Wasm.Instance` objects](#wasminstance-objects)
+* `Wasm.CompileError` : a [NativeError](http://tc39.github.io/ecma262/#sec-nativeerror-object-structure)
+   which indicates an error during WebAssembly decoding or validation
+* `Wasm.RuntimeError` : a [NativeError](http://tc39.github.io/ecma262/#sec-nativeerror-object-structure)
+   which indicates an error while running WebAssembly code
+
+### Function Properties of the `Wasm` object
+
+#### `Wasm.compile`
+
+The `compile` function has the signature:
+```
+Promise<Wasm.Module> compile(TypedArray bytes)
```

Comment:
Sorry, I had missed the case where the point was to make a view on part of a larger ArrayBuffer. I guess the WebGL precedent makes it reasonable to not check types. I hope it's not a very strong need to avoid allocating the TypedArray; that shouldn't be too expensive for this kind of case, I'd imagine, given how much work it'll take to generate machine code as output from the compilation.